### DISCUSSION TITLE
A Few Feature Requests

### DISCUSSION THREAD

**Discussion by benf2004** [stuartthomas25/JParty#6] [Ideas]
Hi! This is such a fun app. My family and I play it all the time. I also had a fun Jeopardy-themed party with some friends. Thanks for your hard work and the recent update is so fun!

I have a few (small) feature requests that I think would improve the app: 

- There should be a toggle to disable the flashing of "space" and the arrow keys for the moderator. I find it quite distracting when trying to read the questions (I've already disabled it for my own local build). 
- I would love if there was a history tracker for games played because we've had a few times where we would start a game and realize we've already played it (this is because we like to play the easy college/teen Jeopardy's haha). I'm thinking when a user types in game id, below the "date/comments", it says "you've played this game x number of times". I understand this may be too complicated to store all the played game IDs (just for J-archive), so no worries if you don't have the time/energy to add this feature. 

I also was wondering if there's a way to run the buzzers on an internet-hosted server (just on a local build where I own the server). The usecase for this is if I don't want to have everyone on my local wifi or play over Zoom. I don't really understand how the Tornado server works, so is there a way to modify the controller.py script to make the websocket work over the internet? 

Lastly, would you be fine if I made an r/Jeopardy post about this app? I think it's cool and I would love to share it.

**Comment by stuartthomas25** [comment]
Hi Ben! Thank you so much for your interest and support! I truly appreciate it.
Regarding you feature requests, I think these are great ideas. Feel free to add them as Issues with the "enhancement" label and I will see what I can do to implement them. 

- For the first, I am thinking of adding a little preferences menu where this could be set, among other things in the future such as the tornado port. 
- I like the history feature, though I'll need to think about how to add some persistent storage to the application. 

Finally, I had considered making this remote compatible but have been a little worried about buzzer latency over the internet. The hacky way to make this happen is to port-forward 8080 to your local machine and then send users to your remote IP address (the QR code would not work in this case). I haven't tried this so if you're interested I would be curious how well it works.

Regarding posting to r/Jeopardy, I have been planning on doing it but only after I have a reliable Windows version. Also I am a little worried about publicizing this because the J-Archive does not allow scraping.
